cells/std/lib/writeShellEntrypoint.nix

```diff
@@ -0,0 +1,212 @@
+        ++ [
+          # prepare a special layer that bundles all generic debugging packages
+          (n2c.buildLayer {
+            deps = debug-tools ++ [debug-banner];
+            maxLayers = 1;
+          })
+          # prepare a special layer that bundles the relatively stable extra debugging packages
+          (n2c.buildLayer {
+            deps = entrypoint.debugInputs;
+            maxLayers = 4;
+          })
+        ];
+      contents = (oci.contents or []) ++ [debug-bin];
+      config =
+        oci.config
+        // {
+          Labels =
+            oci.config.Labels
+            // {
+              "org.opencontainers.image.title" = "Debug Image for ${l.getName entrypoint.package}";
+            };
+        };
+    };
+
+  entrypoint = {
+    # the installable that is wrapped by this entrypoint (re-exported)
+    package,
+    # the bash litteral string of the entrypoint
+    entrypoint,
+    # initialize environment variables with these defaults
+    env ? {},
+    # runtime installables that the entrypoint or liveness/readiness probe uses (re-exported)
+    runtimeInputs ? [],
+    # domain specific debugging utilities (re-exported)
+    debugInputs ? [],
+    # domain specific liveness probe (re-exported)
+    livenessProbe ? null,
+    # domain specific readiness probe (re-exported)
+    readinessProbe ? null,
+  }: let
+    prelude = ''
+      #!${nixpkgs.pkgsStatic.bash.out}/bin/bash
+      set -o errexit
+      set -o nounset
+      set -o pipefail
+
+      export PATH="${l.makeBinPath runtimeInputs}:$PATH"
+
+      ${l.optionalString (nixpkgs.stdenv.hostPlatform.libc == "glibc") "export LOCALE_ARCHIVE=${locales}/lib/locale/locale-archive"}
+      ${l.concatStringsSep "\n" (l.mapAttrsToList (n: v: "export ${n}=${''"$''}{${n}:-${toString v}}${''"''}") env)}
+    '';
+    checkPhase = ''
+      runHook preCheck
+      ${nixpkgs.stdenv.shell} -n $out/bin/entrypoint
+      ${nixpkgs.shellcheck}/bin/shellcheck $out/bin/entrypoint
+      runHook postCheck
+    '';
+    live =
+      if livenessProbe != null
+      then
+        nixpkgs.writeTextFile {
+          name = "live";
+          executable = true;
+          destination = "/bin/live";
+          inherit checkPhase;
+          text = ''
+            ${prelude}
+
+            ${livenessProbe}
+          '';
+        }
+      else null;
+    ready =
+      if readinessProbe != null
+      then
+        nixpkgs.writeTextFile {
+          name = "ready";
+          executable = true;
+          destination = "/bin/ready";
+          inherit checkPhase;
+          text = ''
+            ${prelude}
+
+            ${readinessProbe}
+          '';
+        }
+      else null;
+    inner =
+      nixpkgs.writeTextFile {
+        name = "entrypoint";
+        executable = true;
+        destination = "/bin/entrypoint";
+        inherit checkPhase;
+        text = ''
+          ${prelude}
+
+          [ -n "''${DEBUG_SLEEP:-}" ] && sleep "$DEBUG_SLEEP"
```

Comment:
maybe `sleep "${DEBUG_SLEEP:-0}"` instead? Otherwise using `[[ ... ]]` would avoid the need to call `test` and use the bash builtin instead.